File: webManageProject/src/api/groupType.ts
import request from '@/utils/request'

export interface GroupType {
  id: number
  sportsMeetingId: number
  name: string
  perTeamLimit?: number
  limitEventIds?: string | null
  perPersonLimit?: number
  personLimitEventIds?: string | null
}

export function getGroupTypeList(sportsMeetingId: number) {
  return request.get('/admin/group-type/list', { params: { sportsMeetingId } })
}

export function addGroupType(data: Partial<GroupType>) {
  return request.post('/admin/group-type/add', data)
}

export function updateGroupType(data: Partial<GroupType>) {
  return request.put('/admin/group-type/update', data)
}

export function deleteGroupType(id: number) {
  return request.delete(`/admin/group-type/${id}`)
}

export function getLimitConfig(groupTypeId: number) {
  return request.get('/admin/group-type/limitConfig', { params: { groupTypeId } })
}

export function saveLimitConfig(data: {
  groupTypeId: number
  perTeamLimit: number
  eventIds: number[]
  perPersonLimit: number
  personEventIds: number[]
}) {
  return request.post('/admin/group-type/saveLimitConfig', data)
}

File: SportBackend/src/main/java/com/dlust/sportbackend/entity/GroupType.java
package com.dlust.sportbackend.entity;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class GroupType {
    private Long id;
    private Long sportsMeetingId;
    private String name;
    private Integer perTeamLimit;       // 每代表队在选定项目中限报人数(0=不限)
    private String limitEventIds;       // 每代表队限报选中项目ID数组JSON(如"[1,2,3]")
    private Integer perPersonLimit;     // 每人限报项目数(0=不限)
    private String personLimitEventIds; // 每人限报选中项目ID数组JSON
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}
